contracts/stellar-governance/src/contract.rs

```diff
@@ -0,0 +1,341 @@
+        let proposal_hash = Self::get_proposal_hash(
+            env,
+            target.clone(),
+            call_data.clone(),
+            function,
+            native_value,
+        );
+
+        match command_type {
+            CommandType::ScheduleTimeLockProposal => {
+                let scheduled_eta = TimeLock::schedule_time_lock(env, proposal_hash.clone(), eta)?;
+
+                ProposalScheduledEvent {
+                    target,
+                    eta: scheduled_eta,
+                    proposal_hash,
+                    call_data,
+                }
+                .emit(env);
+            }
+            CommandType::CancelTimeLockProposal => {
+                TimeLock::cancel_time_lock(env, proposal_hash.clone())?;
+
+                ProposalCancelledEvent {
+                    target,
+                    proposal_hash,
+                    call_data,
+                }
+                .emit(env);
+            }
+            CommandType::ApproveOperatorProposal => {
+                storage::set_operator_approval(env, proposal_hash.clone(), &true);
+
+                OperatorProposalApprovedEvent {
+                    target,
+                    proposal_hash,
+                    call_data,
+                }
+                .emit(env);
+            }
+            CommandType::CancelOperatorApproval => {
+                storage::set_operator_approval(env, proposal_hash.clone(), &false);
+
+                OperatorProposalCancelledEvent {
+                    target,
+                    proposal_hash,
+                    call_data,
+                }
+                .emit(env);
+            }
+        }
+
+        Ok(())
+    }
+
+    pub fn get_proposal_hash(
+        env: &Env,
+        target: Address,
+        call_data: Bytes,
+        function: Symbol,
+        native_value: i128,
+    ) -> Bytes {
+        let data = vec![
+            env,
+            target.to_val(),
+            call_data.to_val(),
+            function.to_val(),
+            native_value.into_val(env),
+        ];
+
+        Vec::to_xdr(data, env)
+    }
+
+    fn call_target(
+        env: &Env,
+        target: &Address,
+        function: &Symbol,
+        call_data: &Bytes,
+        native_value: Option<i128>,
+        token_address: &Address,
+    ) -> Result<Val, ContractError> {
+        if let Some(value) = native_value {
+            if value > 0 {
+                let token_client = soroban_sdk::token::Client::new(env, token_address);
+                let balance: i128 = token_client.balance(&env.current_contract_address());
+
+                if balance < value {
+                    return Err(ContractError::InsufficientBalance);
+                }
+
+                token_client.transfer(&env.current_contract_address(), target, &value);
+            }
+        }
+        let args = if !call_data.is_empty() {
+            vec![env, call_data.to_val()]
+        } else {
+            Vec::new(env)
+        };
```

Comment:
the other way would be more natural?

```suggestion
        let args = if call_data.is_empty() {
            Vec::new(env)
        } else {
            vec![env, call_data.to_val()]
        };
```